python_modules/libraries/dagster-dlt/dagster_dlt/components/dlt_load_collection/scaffolder.py

```diff
@@ -49,6 +50,40 @@ def _extract_pipeline_and_source_from_init_file(
     return PipelineAndSource(check.not_none(pipeline_src), check.not_none(source_src))
 
 
+def _nested_dict_to_dot_separated_keys(config: dict) -> dict:
+    """Convert a nested dictionary into a flattened dictionary with dot-separated keys.
+
+    Args:
+        config: A dictionary that may contain nested dictionaries
+
+    Returns:
+        A flattened dictionary where nested keys are joined with dots
+    """
+    config_dot_separated = {}
+    for key, value in config.items():
+        if isinstance(value, dict):
+            nested = _nested_dict_to_dot_separated_keys(value)
+            for k, v in nested.items():
+                config_dot_separated[f"{key}.{k}"] = v
+        else:
+            config_dot_separated[key] = value
+    return config_dot_separated
+
+
+def _extract_env_vars_from_dlt_config(config_file: Path) -> dict:
+    from dlt.common.configuration.providers.environ import EnvironProvider
```

Comment:
add a little comment section here about what we are assuming ie which config file what structure is expected etc